Client/Telemetry/TelemetryWrapper.swift

```diff
@@ -510,8 +510,13 @@ extension TelemetryWrapper {
         case (.action, .tap, .goToSettingsDefaultBrowserOnboarding, _, _):
             GleanMetrics.DefaultBrowserOnboarding.goToSettingsPressed.add()
         // Onboarding
-        case (.action, .press, .dismissedOnboarding, let slideNum, _):
-            GleanMetrics.Onboarding.finish.record(extra: [.slideNum : slideNum])
+        case (.action, .press, .dismissedOnboarding, _, let extras):
+            if let slideNum = extras?["slide-num"] as? Int32 {
+                GleanMetrics.Onboarding.finish.record(GleanMetrics.Onboarding.FinishExtra(slideNum: slideNum))
```

Comment:
Ref https://mozilla.github.io/glean/book/reference/metrics/event.html#recording-api for this new API.